packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -80,12 +80,50 @@ export 'package:flutter/services.dart' show
 
 // BIDIRECTIONAL TEXT SUPPORT
 
+/// An [InheritedElement] that has hundreds of dependencies but will
+/// infrequently change.  This provides a performance tradeoff where building
+/// the [Widget]s is faster but performing updates is slower.
+class _UbiquitousInheritedElement extends InheritedElement {
+  /// Creates an element that uses the given widget as its configuration.
+  _UbiquitousInheritedElement(InheritedWidget widget) : super(widget);
+
+  @override
+  void setDependencies(Element dependent, Object? value) {
+    if (value != null) {
+      throw Exception('Setting aspects on _UbiquitousInheritedElement is not supported.');
+    }
+  }
+
+  @override
+  Object? getDependencies(Element dependent) {
+    return null;
+  }
+
+  @override
+  void notifyClients(InheritedWidget oldWidget) {
+    visitChildren((final Element dependent) {
+      notifyDependent(oldWidget, dependent);
```

Comment:
I could make this a bit more surgical but I'd have to change the public interface for InheritedElement.

Reply:
I had to make this more surgical in the end since `Directionality` is used for tests that look for redundant rebuilding.